### ISSUE TITLE
No browser refresh when changing HTML resources

### ISSUE THREAD

**Issue by julienguiraud** [jhipster/generator-jhipster#1900]
After installing a new app, running grunt serve and changing audit.html, the browser  does not refresh.

When we insert the path `'src/main/webapp/scripts/**/*.html` in `browserSync.dev.bsFiles.src` after line 83, that works but only after relaunching `grunt serve`.

Windows 7 64 bits
JHipster : 2.19.0
Chrome, FF last versions
node : 0.12.6
npm : 2.11.2

Issues raised in 4 differents PC with almost same configuration.

Thanks

**Comment by andidev** [comment]
If you are using firefox try doing this:
https://support.mozilla.org/sv/questions/905902

If you are using chrome try this:
http://stackoverflow.com/questions/5690269/disabling-chrome-cache-for-website-development

did it solve your issue?

**Comment by julienguiraud** [comment]
Thanks for your answer but I don't want to change my browser settings.

I did not try your solution also because I think it won't solve my problem for these two reasons:
* index.html is correcly refreshed.
* the workaround described above works.

That's why I think that it is not a cache problem but perhaps I am wrong...

**Comment by andidev** [comment]
Sorry must have read this issue to fast. So are you saying browser sync is not configured to sync files in the script folder?

**Comment by andidev** [comment]
Shouldn't the already existing configuration `'src/main/webapp/**/*.html',` on [line 96](https://github.com/jhipster/generator-jhipster/blob/master/app/templates/Gruntfile.js#L96) handle the refresh of html files in the script folder?

This sounds rather like a browserSync issue than a jhipster issue.

**Comment by andidev** [comment]
So I can't reproduce this. Browser refresh of the `audit.html` file works for me in both firefox and chrome.
Though I'm sitting on OSX.

This is most certainly an browserSync issue since the configuration seems to be correct.

**Comment by julienguiraud** [comment]
You are right. I just posted an issue on BrowserSync project.

Thanks for your support.

**Comment by andidev** [comment]
I am having trouble now as well with the entity pages. Though my browserSync seems to refresh the page but no changes are visible. However refreshing the page on port 8080 (non broswer sync) seems to work fine.

I also noticed that using the ip address instead of localhost:3000 seems to work.

I feel I'm going slightly mad, nothing makes any sense anymore.

**Comment by deepu105** [comment]
I dont know if its just me who thinks chrome doesnt obey cache rules, it
doesn't refresh files even with a cntrl+f5 i have to keep the console open
with caching disabled all the time. Any one else experience the same?

**Comment by andidev** [comment]
I always keep the console open so I wouldn't know. Did not now about ctrl f5.
Though I'm also having issues with chrome sometimes crashing with bluescreens after a resresh. Are you experiencing that as well @deepu105?

**Comment by deepu105** [comment]
No blue screens just the cache issue

**Comment by bonhamcm** [comment]
I've had the same issue with Browsersync not updating the HTML entity files with Chrome on OS X. Sometimes even hard-clearing the cache doesn't work. I wonder if it's the AngularJS template cache?

**Comment by kevintvh** [comment]
I got same issue. My workaround is run "grunt serve" and working on it until getting caching issue. Restart "grunt serve" again when page does not refresh. 

Refreshing the page on port 8080 does not work sometimes.

**Comment by andidev** [comment]
So I think the issues I am having is caused by ngCacheBuster.
Not sure if we all share the same issue but this is my issue:
1. I open e.g. an entity page
2. I change the html source
3. BrowserSync refreshes the page
4. The changes made in html source is not displayed

I solved the issue by disabling ngCacheBuster by commenting the following line in app.js
```javascript
        httpRequestInterceptorCacheBusterProvider.setMatchlist([/.*api.*/, /.*protected.*/], true);
```
the browser is now refreshed with the changes made in html source.

And when I now re-enable ngCacheBuster again the browser gets refreshed without the changes again.

However this bug does not always appear and it sometimes works with ngCacheBuster.

@jdubois 
How much value does ngCacheBuster add? According to the homepage it is `Especially useful with Internet Explorer (IE8, IE9)` which makes me think that it may be worth removing?

@julienguiraud, @kevintvh, @bonhamcm 
Are we having the same issue? Does disabling ngCacheBuster solve your issues?

**Comment by jdubois** [comment]
Yes cacheBuster is only here for older IE versions. As time goes by, of course, it is less and less interesting to have it.
I don't really like to have this on every request, but that was the lesser of 2 evils.
Now, if it really does trigger that bug, then it's another story, and in that case we should remove it -> we need to check if IE9 is really still a requirement for many people

**Comment by julienguiraud** [comment]
@andidev With last version of Chrome, the issue is not resolved by disabling ngCacheBuster.

**Comment by jdubois** [comment]
I just tested and I have the same issue!! I can't believe this!!!

I have no idea what causes this, excepted that this used to work very well, not so long ago.

This is a top priority issue, definitely.

**Comment by jdubois** [comment]
So I've upgraded BrowserSync to the latest version (see https://github.com/jhipster/generator-jhipster/commit/9596edb46c8d4e6bca40018fa7e804885195e3b0)

The issue is still present, but it's going on and off randomly. The only solution is to [disable the cache in dev tools](http://stackoverflow.com/questions/5690269/disabling-chrome-cache-for-website-development) like suggested @andidev 

As I'm always working with the dev tools opened, I might not have noticed this was buggy since a long time.

cc @PeterEltgroth @amr -> you might need to also upgrade our BrowserSync support for Gulp

This is indeed a BrowserSync issue, so let's wait until they solve their issue https://github.com/BrowserSync/grunt-browser-sync/issues/120

I'm closing this as this is not a JHipster issue, and as we have a workaround.

**Comment by jdubois** [comment]
Great news, BrowserSync has confirmed this is a bug and that it will be corrected soon!

**Comment by deepu105** [comment]
I stopped using this long back, coz chrome dev tools workspace is way
better, dont know how many of you have tried it. Just add the webapp folder
of the app to workspace and voila you can edit js and save right in the
console during a debug cycle and automatically reloads the changes you do
as well

**Comment by jrSin2** [comment]
You need to update jhiTranslate file under src/main/webapp/i18n/en/
